### PR TITLE
Create Freshdesk plugin with destinations for contacts and companies

### PR DESCRIPTION
**Pull request by aelafifi**
## Change description

- Create an app for Freshdesk.
- Create a destination for Contacts.
- Create a destination for Companies.

### Development

- [x] Application changes have been tested appropriately

### Impact

- [x] Code follows company security practices and guidelines
- [x] Security impact of change has been considered
- [x] Performance impact of change has been considered
- [x] Possible migration needs considered (model migrations, config migrations, etc.)

Please explain any security, performance, migration, or other impacts if relevant:

### Code review

- [x] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached where applicable.
- [x] Relevant tags have been added to the PR (bug, enhancement, internal, etc.)

### CODE REVIEW

File: plugins/@grouparoo/freshdesk/__tests__/utils/shared.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import { helper } from "@grouparoo/spec-helper";
+
+export async function indexUsers(isNewNock: boolean) {
+  // wait for search index
+  if (isNewNock) {
+    await helper.sleep(helper.shortTime);
```
Comment (bleonard) [inline plugins/@grouparoo/freshdesk/__tests__/utils/shared.ts]: 👍

File: plugins/@grouparoo/freshdesk/src/lib/client.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+type ContactSearchParams = "email" | "phone" | "mobile" | "unique_external_id";
+
+export class FreshdeskClient {
+  readonly axios: AxiosInstance;
+
+  constructor(readonly domainName: string, apiKey: string) {
+    this.axios = axios.create({
+      baseURL: domainName,
+      auth: { username: apiKey, password: "X" },
```
Comment (bleonard) [inline plugins/@grouparoo/freshdesk/src/lib/client.ts:11]: This `X` is odd. That the spec?

File: plugins/@grouparoo/freshdesk/src/lib/client.ts [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+      return null;
+    }
+
+    const response = await this.axios.get("/api/v2/companies/autocomplete", {
+      params: { name },
+    });
+
+    return (
+      response.data?.companies?.find(
+        (company) => company.name.trim() === name.trim()
```
Comment (bleonard) [inline plugins/@grouparoo/freshdesk/src/lib/client.ts]: not sure if we should normalize case or not here

File: plugins/@grouparoo/freshdesk/src/lib/export-companies/exportRecord.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+  if (Object.keys(newRecordProperties).length === 0) {
+    return { success: true };
+  }
+
+  if (!newRecordProperties["Name"]) {
+    throw new Error(`newRecordProperties[Name] is a required mapping`);
+  }
+
+  const oldName = oldRecordProperties["Name"];
+  const newName = newRecordProperties["Name"];
```
Comment (bleonard) [inline plugins/@grouparoo/freshdesk/src/lib/export-companies/exportRecord.ts:23]: saves an API call if we check these are different

**Review by bleonard — COMMENTED** [review]
Looking good!

File: plugins/@grouparoo/freshdesk/src/lib/client.ts (hunk 2 above)
Comment (aelafifi) [reply plugins/@grouparoo/freshdesk/src/lib/client.ts:11]: Yeah! It's really odd, but that's how they documented it and how it works!

File: plugins/@grouparoo/freshdesk/src/lib/client.ts (hunk 3 above)
Comment (aelafifi) [reply plugins/@grouparoo/freshdesk/src/lib/client.ts]: I think we should.

**Review by bleonard — APPROVED** [review]

**Pull request merged**